File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/EnvironmentKeys.java
package uk.nhs.nhsx.core;

import uk.nhs.nhsx.core.Environment.EnvironmentKey;
import uk.nhs.nhsx.core.aws.s3.BucketName;

import static uk.nhs.nhsx.core.Environment.EnvironmentKey.*;

public class EnvironmentKeys {

    public static final EnvironmentKey<BucketName> SUBMISSION_BUCKET_NAME = value("SUBMISSION_BUCKET_NAME", BucketName::of);
    public static final EnvironmentKey<BucketName> BUCKET_NAME = value("BUCKET_NAME", BucketName::of);
    public static final EnvironmentKey<String> DISTRIBUTION_ID = string("DISTRIBUTION_ID");
    public static final EnvironmentKey<String> DISTRIBUTION_INVALIDATION_PATTERN = string("DISTRIBUTION_INVALIDATION_PATTERN");
    public static final EnvironmentKey<String> SUBMISSIONS_TOKENS_TABLE = string("submission_tokens_table");
    public static final EnvironmentKey<BucketName> SUBMISSION_STORE = value("SUBMISSION_STORE", BucketName::of);
    public static final EnvironmentKey<String> SSM_CIRCUIT_BREAKER_BASE_NAME = string("SSM_CIRCUIT_BREAKER_BASE_NAME");
}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/core/EnvironmentKeys.kt
package uk.nhs.nhsx.core

import uk.nhs.nhsx.core.Environment.EnvironmentKey.Companion.string
import uk.nhs.nhsx.core.Environment.EnvironmentKey.Companion.value
import uk.nhs.nhsx.core.aws.dynamodb.TableName
import uk.nhs.nhsx.core.aws.s3.BucketName

object EnvironmentKeys {
    val SUBMISSION_BUCKET_NAME = value("SUBMISSION_BUCKET_NAME", BucketName)
    val BUCKET_NAME = value("BUCKET_NAME", BucketName)
    val DISTRIBUTION_ID = string("DISTRIBUTION_ID")
    val DISTRIBUTION_INVALIDATION_PATTERN = string("DISTRIBUTION_INVALIDATION_PATTERN")
    val SUBMISSIONS_TOKENS_TABLE = value("submission_tokens_table", TableName)
    val SUBMISSION_STORE = value("SUBMISSION_STORE", BucketName)
    val SSM_CIRCUIT_BREAKER_BASE_NAME = string("SSM_CIRCUIT_BREAKER_BASE_NAME")
}
